docs/automation.md

```diff
@@ -0,0 +1,133 @@
+# Automation
```

Comment:
I feel like this doc is not about `Automation`, but `How to operate workflows though Argo APIs`.

Reply:
This doc is intended to focus on automation with Argo Workflows. I think each section could explain the specific thing you are doing with it. I'll ponder a bit.